test/intersectionWith.js

```diff
@@ -10,4 +10,16 @@ describe('intersectionWith', function() {
     eq(R.intersectionWith(eqA, Ro, So), [{a: 3}, {a: 4}]);
   });
 
+  it('only returns elements from the first list', function() {
```

Comment:
if this is the contract, then the name *must* change. `innerJoin` is a better fit for this behavior.